setup.py

```diff
@@ -1,24 +1,14 @@
 from setuptools import setup
 
+install_requires =[]
```

Comment:
Actually your own comment says what's wrong with your code: "When deciding to use the more efficient list comp vs. for-loop - if there's a side-effect, pick for-loop, otherwise when just accumulating a result, use list-comp."
Appending to the list of interest inside the comprehension *is* a side-effect.